From 9f74c099d8eb6ce1c9fe8b593fc8a555b2137112 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 16 Mar 2020 06:27:24 +0200
Subject: [PATCH] [SW-0] hl: update spi offsets per device

The size of BTL which lays at the begin of flash is per device
This fact impacts layout of the other images
No more common SPI offsets for preboot/uboot/...

Change-Id: Ide9ec52db3233936a2b407f1acbb125218b3f673
---

diff --git a/app/preboot.c b/app/preboot.c
index a59076e..75a6330 100644
--- a/app/preboot.c
+++ b/app/preboot.c
@@ -13,7 +13,7 @@
 #include <hl_fw_if.h>
 #include <hl_boot_if.h>
 
-
+#include "common.h"
 #include "preboot.h"
 #include "uart.h"
 #include "spi.h"
diff --git a/gaudi/common.h b/gaudi/common.h
index a9fe4d5..cbeb9eb 100644
--- a/gaudi/common.h
+++ b/gaudi/common.h
@@ -7,6 +7,11 @@
 
 #include <preboot.h>
 
+#define HL_SPI_NEXT_IMAGE_OFF	0x60000
+#define HL_SPI_MODIFI_IMAGE_OFF	0x60004
+#define HL_SPI_UBOOT_OFF	0xB0000
+#define HL_SPI_PREBOOT_OFF	0x1A0000
+
 #define PPBOOT_TEXT_BASE		0x7FF0000000
 #define PREBOOT_TEXT_BASE		(PPBOOT_TEXT_BASE + PPBOOT_SIZE)
 
diff --git a/goya/common.h b/goya/common.h
index 308d933..e985bb0 100644
--- a/goya/common.h
+++ b/goya/common.h
@@ -7,6 +7,11 @@
 
 #include <preboot.h>
 
+#define HL_SPI_NEXT_IMAGE_OFF	0x60000
+#define HL_SPI_MODIFI_IMAGE_OFF	0x60004
+#define HL_SPI_UBOOT_OFF	0xB0000
+#define HL_SPI_PREBOOT_OFF	0x1A0000
+
 #define PPBOOT_TEXT_BASE	0x7FF0000000
 #define PREBOOT_TEXT_BASE	(PPBOOT_TEXT_BASE + PPBOOT_SIZE)
 
diff --git a/goya2/common.h b/goya2/common.h
index a9fe4d5..fc28d4b 100644
--- a/goya2/common.h
+++ b/goya2/common.h
@@ -7,6 +7,11 @@
 
 #include <preboot.h>
 
+#define HL_SPI_NEXT_IMAGE_OFF	0x80000
+#define HL_SPI_MODIFI_IMAGE_OFF	0x80004
+#define HL_SPI_UBOOT_OFF	0xD0000
+#define HL_SPI_PREBOOT_OFF	0x1C0000
+
 #define PPBOOT_TEXT_BASE		0x7FF0000000
 #define PREBOOT_TEXT_BASE		(PPBOOT_TEXT_BASE + PPBOOT_SIZE)
 
diff --git a/include/preboot.h b/include/preboot.h
index 889a0d0..33b04d2 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -31,10 +31,6 @@
 #define UBOOT_SIZE		0xF0000		/* 960KB */
 #define HL_UART_BAUDRATE	115200
 #define HL_SPI_FLASH_MAX_FREQ	25000000
-#define HL_SPI_NEXT_IMAGE_OFF	0x60000
-#define HL_SPI_MODIFI_IMAGE_OFF	0x60004
-#define HL_SPI_UBOOT_OFF	0xB0000
-#define HL_SPI_PREBOOT_OFF	0x1A0000
 #define HL_SPI_PRIMARY_BASE	0
 #define HL_SPI_SECONDARY_BASE	0x800000
 
